### PR TITLE
[Wait for #3390] [Test] Unittest for filter shared key @open sesame 9/23 14:20

### PR DESCRIPTION
**Pull request by kimjh12**
Unittest to validate filter with shared key

Signed-off-by: Junhwan Kim <jejudo.kim@samsung.com>

---
**Self evaluation:**
1. Build test: [*]Passed [ ]Failed [ ]Skipped
2. Run test: [*]Passed [ ]Failed [ ]Skipped

### CODE REVIEW

File: tests/nnstreamer_filter_shared_model/unittest_filter_shared_model.cc [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+  g_free (new_model_path);
+  g_free (path);
+
+  EXPECT_EQ (setPipelineStateSync (pipeline, GST_STATE_PLAYING, UNITTEST_STATECHANGE_TIMEOUT), 0);
+  g_usleep (TEST_DEFAULT_SLEEP_TIME);
+  EXPECT_EQ (setPipelineStateSync (pipeline, GST_STATE_PAUSED, UNITTEST_STATECHANGE_TIMEOUT), 0);
+  g_usleep (TEST_DEFAULT_SLEEP_TIME);
+
+  /* same output with new model */
+  EXPECT_NE (res[0], 0);
```
Comment (anyj0527) [inline tests/nnstreamer_filter_shared_model/unittest_filter_shared_model.cc:196]: signed vs unsigned issue: 0U

File: tests/meson.build [hunk 2]
```diff
@@ -246,6 +246,17 @@ if gtest_dep.found()
   if get_option('enable-openvino')
     subdir('nnstreamer_filter_openvino')
   endif
+
+  if tflite_support_is_available
+    sunittest_filter_shared_model = executable('unittest_filter_shared_model',
```
Comment (zhoonit) [inline tests/meson.build:251]: I wonder about the variable `sunittest` may be typo?

File: tests/nnstreamer_filter_shared_model/unittest_filter_shared_model.cc [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+      "tensor_converter ! tensor_filter name=filter2 framework=tensorflow-lite model=%s is-updatable=TRUE "
+      "shared-tensor-filter-key=%s ! tensor_sink name=sink2",
+      image_path, model_path1, shared_key, model_path1, shared_key);
+  g_free (root_path);
+  g_free (model_path1);
+  g_free (image_path);
+
+  pipeline = gst_parse_launch (pipeline_str, NULL);
+
+  EXPECT_NE (setPipelineStateSync (pipeline, GST_STATE_PLAYING, UNITTEST_STATECHANGE_TIMEOUT), 0);
```
Comment (zhoonit) [inline tests/nnstreamer_filter_shared_model/unittest_filter_shared_model.cc:133]: we will need to match the golden test data with this strategy, but I guess someone else could help this :)

**Review by zhoonit — APPROVED** [review]
LGTM :)

**Comment by myungjoo** [comment]
#3918 will handle this.

**Pull request closed without merging**